src/Refactoring-UI/RBRenameVariableDriver.class.st

```diff
@@ -0,0 +1,88 @@
+"
+I am an interaction class to rename both instance and class variables
+"
+Class {
+	#name : 'RBRenameVariableDriver',
+	#superclass : 'RBInteractionDriver',
+	#instVars : [
+		'variable',
+		'newName',
+		'oldName',
+		'shouldEscape'
+	],
+	#category : 'Refactoring-UI-Drivers',
+	#package : 'Refactoring-UI',
+	#tag : 'Drivers'
+}
+
+{ #category : 'class name' }
+RBRenameVariableDriver class >> rename: aClyVariable [
+
+	^ self new
+		  variable: aClyVariable;
+		  yourself
+]
+
+{ #category : 'execution' }
+RBRenameVariableDriver >> classRefactoring [
+	^ RBRenameInstanceVariableRefactoring
+]
+
+{ #category : 'resources' }
+RBRenameVariableDriver >> configureRefactoring [
+
+	refactoring := variable createVariableRefactoring:
+		               self classRefactoring
+]
+
+{ #category : 'accessing' }
+RBRenameVariableDriver >> model: aRBBrowserEnvironment [ 
+	model := aRBBrowserEnvironment
+]
+
+{ #category : 'execution' }
+RBRenameVariableDriver >> requestNewNameBasedOn: aName [
+
+	newName := SpRequestDialog new
+		           title: 'Please provide a new variable name';
+		           label: 'The new name should be unique among the variables in the class hierarchy';
+		           text: aName;
+		           openModal.
+
+	newName ifNil: [
```

Comment:
We should also check that the name is valid.

Reply:
according to our last meeting, for the moment we will only validate after setting the newName to the refactoring